azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/internal/BaseAzureComputeApiLiveTest.java

```diff
@@ -74,6 +83,17 @@ protected String getEndpoint() {
       return endpoint;
    }
 
+   protected String getSubscriptionId() {
+      String subscriptionid = null;
+      String endpoint = null;
+      endpoint = getEndpoint();
+      if (endpoint != null) {
+         subscriptionid = endpoint.substring(endpoint.lastIndexOf("/") + 1);
+      }
+      assertNotNull(subscriptionid);
+      return subscriptionid;
+   }
```

Comment:
If this method is no longer needed, then remove it. If it is needed, then change it to get the subscriptionId from the injector, now that it is bound to the context:
```java
return injector.getInstance(Key.get(String.class, Names.named("subscriptionId")));
```
You can initialize the injector into a protected variable in the base test class. You can look at the [BaseDigitalOcean2ApiLiveTest](https://github.com/jclouds/jclouds/blob/master/providers/digitalocean2/src/test/java/org/jclouds/digitalocean2/internal/BaseDigitalOcean2ApiLiveTest.java) for an example.